src/writer.js

```diff
@@ -19,7 +19,13 @@ class Writer {
   }
 
   append (span) {
-    const trace = span.context()._trace
+    const spanContext = span.context()
+    const trace = spanContext._trace
+
+    if (spanContext._sampling !== undefined && spanContext._sampling.pleaseDrop === true) {
+      log.debug(() => `Dropping span due to user configured filtering: ${span}`)
```

Comment:
Not sure this log should happen for every individual span as it could be confusing for child spans.

Reply:
Makes sense, I'll only print if the span has no parent? Or we could be silent

Reply:
> I'll only print if the span has no parent

That would work. There are a few cases where it may not work properly if the scope is lost for any reason but that's an edge case, and I prefer to have the logging and accept this limitation.

Reply:
Took a slightly different approach here, using the `trace.started.length === trace.finished.length` check

Reply:
The issue with that is it's possible that a child starts after the trace is finished which will effectively start a new local trace but with the other trace as parent. However, since the agent has the same behavior for `ignore_resouces` I think it's acceptable and probably even better in a way to be consistent.